imod/tests/test_mf6/test_mf6_wel.py

```diff
@@ -137,35 +141,141 @@ def test_to_mf6_pkg__high_lvl_transient(basic_dis, well_high_lvl_test_data_trans
     np.testing.assert_equal(mf6_ds["rate"].values, rate_expected)
 
 
-def test_clip_box__high_lvl_stationary(well_high_lvl_test_data_stationary):
+class ClipBoxCases:
+    @staticmethod
+    def case_clip_xy(parameterizable_basic_dis):
+        clip_arguments = {
+            "x_min": 52.0,
+            "x_max": 76.0,
+            "y_max": 67.0,
+        }
+
+        expected_dims = {"index": 3, "species": 2}
+        return clip_arguments, expected_dims, does_not_raise()
+
+    @staticmethod
+    def case_clip_layer_max(parameterizable_basic_dis):
+        _, top, bottom = parameterizable_basic_dis
+        clip_arguments = {"layer_max": 2, "bottom": bottom, "top": top}
+
+        expected_dims = {"index": 4, "species": 2}
+        return clip_arguments, expected_dims, does_not_raise()
+
+    @staticmethod
+    def case_clip_layer_min(parameterizable_basic_dis):
+        _, top, bottom = parameterizable_basic_dis
+        clip_arguments = {"layer_min": 5, "bottom": bottom, "top": top}
+
+        expected_dims = {"index": 4, "species": 2}
+        return clip_arguments, expected_dims, does_not_raise()
+
+    @staticmethod
+    def case_clip_layer_min_layer_max(parameterizable_basic_dis):
+        _, top, bottom = parameterizable_basic_dis
+        clip_arguments = {"layer_min": 1, "layer_max": 1, "bottom": bottom, "top": top}
+
+        expected_dims = {"index": 4, "species": 2}
+        return clip_arguments, expected_dims, does_not_raise()
+
+    @staticmethod
+    def case_clip_top_is_scalar(parameterizable_basic_dis):
+        _, _, bottom = parameterizable_basic_dis
+        top = 0.0
+        clip_arguments = {"layer_max": 2, "bottom": bottom, "top": top}
+
+        expected_dims = {"index": 4, "species": 2}
+        return clip_arguments, expected_dims, does_not_raise()
+
+    @staticmethod
+    def case_clip_top_is_non_layered_griddataarray(parameterizable_basic_dis):
+        _, top, bottom = parameterizable_basic_dis
+        top = top.isel(layer=0).drop_vars("layer")
```

Comment:
I don't see top here defined as griddataarray, missing the call to ``broadcast_to_full_domain here``.